### PR TITLE
[CI] Enable Codecov on PRs

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

It enable Codecov on PRs.

### Motivation

Know when we decrease our coverage.

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+        threshold: 0%
+        # Will set the status to success even if the CI fails
+        if_ci_failed: success
 
+# See https://docs.codecov.com/docs/ignoring-paths
 ignore:
   - "pkg/trace/testutil"  # code only used in tests
-  - "pkg/trace/log"       # logging utilities with no customer facing impact. Nothing to test.
\ No newline at end of file
+  - "pkg/trace/log"       # logging utilities with no customer facing impact. Nothing to test.
```
Comment (CelianR) [inline .codecov.yml:27]: **👏 praise:** ‏

**Review by CelianR — APPROVED** [review]

File: .codecov.yml (hunk 1 above)
Comment (amenasria) [reply .codecov.yml:27]: Ahahahah 🫰

**Pull request merged**